Sources/Authgear.swift

```diff
@@ -573,7 +573,7 @@ public class Authgear {
             }
         }
 
-        DispatchQueue.main.async {
+        DispatchQueue.main.sync {
```

Comment:
I think this could cause deadlock if persistSession was called inside main thread.
Should fix by accept a new `handler: VoidCompletionHandler` parameter in `persistSession` and pass result with that handler.